### ISSUE TITLE
Model for Fine-grained POS Tagging of German Tweets

### ISSUE THREAD

**Issue by stefan-it** [zalandoresearch/flair#51]
Hi,

recently I received the training, dev and test data from the "Fine-grained POS Tagging of German Tweets" paper (which can be found [here](https://pdfs.semanticscholar.org/82c9/90aa15e2e35de8294b4a721785da1ede20d0.pdf)). The highest accuracy mentioned in that paper is 89.42%.

I trained a model with German fasttext embeddings and forward/backward lm embeddings and after 40 epochs (on CPU) an accuracy ~ 92% could be achieved. I double checked that result with an own evaluation script that loads the trained model, predicts the tags and compares the predicted tags with the gold standard and this also achieves 92%.

So I would like to ask if the core team is interested in such a pos tagging model for German twitter data? I would really like to share the model, but then I have to ask the paper author for a permission to share a trained model.

**Comment by alanakbik** [comment]
Hi Stefan, 

wow that's great! Did the training terminate after the annealing rate leveled off or did you set a fixed set of epochs? Could you also post the training parameters? 

We're absolutely interested in including this if possible. The easiest would be to report the results on the comparative results table in the README, but we could also think about packaging the model to that it becomes loadable as a standard pre-trained model. Do you think a model trained over this relatively small dataset would be helpful to people, or do you think the standard POS model for German is enough?

**Comment by stefan-it** [comment]
Hi Alan,

I wanted to apply the `de-pos` model on the test data, but then I recognized, that the tag set differs (STTS vs Universal Tagset). The UD dataset includes the STTS tags in the column next to the universal pos tag, so I'm currently retraining a model. I think we then have a good comparison. 

Btw: the Twitter dataset comes with additional pos tags (support for hashtags, emoticons...)

**Comment by tabergma** [comment]
@stefan-it We would be interested in adding your model to the next flair release. Could you share the model with us? Many thanks in advance!

**Comment by stefan-it** [comment]
@tabergma I did some hyper parameter search with the newly introduces `hyperopt` interface and the accuracy is now 93.06 :)

The model can be found here:

```bash
wget https://schweter.eu/cloud/flair-models/de-pos-twitter-v0.1.pt
```

I will update my README [here](https://github.com/stefan-it/flair-experiments/tree/master/pos-twitter-german) soon. The model was trained with PyTorch in version 1.0 with CUDA 10.

Feel free to include the model in the upcoming release :)
